cva6/tests/custom/sv32/vm_invalid_pte01.S

```diff
@@ -0,0 +1,238 @@
+    sw t3, 8(t1)                                                            # store the mcause value of expected cause
+    la a1,rvtest_data                                                       # loads the address of label rvtest_data
+
+    lw t1,12(a1)
+    bne t1, x0,PTE_LEVEL1_USER_EXECUTE 
+    sw t4,12(a1)
+    
+PTE_LEVEL1_SUPERVISOR_EXECUTE:
+
+# -------------LEVEL 1 PTE Setup for execute test----------------------
+                                                                            # Setup a new PTE to test execute 
+    la a1,check_execute                                                     # loads the address of label vm_en
+    mv a0, a1 
+    ori a2, x0, ( PTE_D | PTE_A | PTE_X | PTE_W | PTE_R )                   # sets the permission bits
+    PTE_SETUP_RV32(a1, a2, t1, a0, LEVEL1)                                  # setup the PTE for level1
+
+# ----------------Set the SATP and change the mode---------------------
+
+    SATP_SETUP_SV32(pgtb_l1)                                                # set the SATP for virtualization
+    la a1,check_execute                                                     # loads the address of check_execute
+    CHANGE_T0_S_MODE(a1)                                                    # changes mode M to S and set the MEPC 
+
+PTE_LEVEL1_USER_EXECUTE:
+
+# -------------LEVEL 1 PTE Setup for execute test----------------------
+                                                                            # Setup a new PTE to test execute 
+    la a1,check_execute                                                     # loads the address of label vm_en
+    mv a0, a1 
+    ori a2, x0, ( PTE_D | PTE_A | PTE_U | PTE_X | PTE_W | PTE_R )           # sets the permission bits
+    PTE_SETUP_RV32(a1, a2, t1, a0, LEVEL1)                                  # setup the PTE for level1
+
+# ----------------Set the SATP and change the mode---------------------
+
+    SATP_SETUP_SV32(pgtb_l1)                                                # set the SATP for virtualization
+    la a1,check_execute                                                     # loads the address of check_execute
+    CHANGE_T0_U_MODE(a1)                                                    # changes mode M to S and set the MEPC 
+
+
+check_execute:
+    li t1, 0x45                                                             # page fault should raise 
+   
+    la a1, rvtest_data
+    lw t1,8(a1)
+    bne t1, x0,exit 
+    sw t4,8(a1)
+
+
+PTE_LEVEL1_USER:
+
+# ----------------LEVEL 1 PTE Setup for load and store test------------
+
+    la a1,vm_en                                                             # loads the address of label vm_en
+    mv a0, a1                                                               # generrates the VA for label vm_en
+    ori a2, x0, ( PTE_D | PTE_A | PTE_U | PTE_X | PTE_W | PTE_R | PTE_V )   # sets the permission bits
```

Comment:
Can pass this PTE_U as an argument to the test? like a define.
for example 
```
#ifdef U_MODE
SET_PTE_U 1
#else
SET_PTE_U 0
```
This way we can reuse the whole code for u mode too

Reply:
Done